src/main/scala/com/fulcrumgenomics/umi/ExtractUmisFromBam.scala

```diff
@@ -121,15 +132,34 @@ class ExtractUmisFromBam
   // Validate the molecular index tags against the read structures and then split them out
   private val (molecularIndexTags1, molecularIndexTags2) = molecularIndexTags match {
     case Seq(one) =>
-      val t1 = Seq.fill(rs1.molecularBarcodeSegments.length)(one)
-      val t2 = Seq.fill(rs2.map(_.molecularBarcodeSegments.length).getOrElse(0))(one)
+      val t1 = Seq.fill(rs1.molecularBarcodeSegments.length.min(1))(one)
+      val t2 = Seq.fill(rs2.map(_.molecularBarcodeSegments.length).getOrElse(0).min(1))(one)
       (t1, t2)
     case tags =>
       // Validate that the # of molecular barcodes in the read structure matches the tags
       val total = rs1.molecularBarcodeSegments.length + rs2.map(_.molecularBarcodeSegments.length).getOrElse(0)
       validate(tags.length == total, s"Read structures contains $total molecular barcode segments, but ${tags.length} tags provided.")
       (tags.take(rs1.molecularBarcodeSegments.length), tags.drop(rs1.molecularBarcodeSegments.length))
   }
+  // Same validation for molecularIndexTagsQuals
+  private val (molecularIndexTagsQuals1, molecularIndexTagsQuals2) = molecularIndexTagsQuals match {
+    case Seq(one) =>
+      val t1 = Seq.fill(rs1.molecularBarcodeSegments.length.min(1))(one)
+      val t2 = Seq.fill(rs2.map(_.molecularBarcodeSegments.length).getOrElse(0).min(1))(one)
+      (t1, t2)
+    case tags =>
+      // Validate that the # of molecular barcodes in the read structure matches the tags
+      if(tags.length > 0) 
+      {
+        val total = rs1.molecularBarcodeSegments.length + rs2.map(_.molecularBarcodeSegments.length).getOrElse(0)
+        validate(tags.length == total, s"Read structures contains $total molecular barcode segments, but ${tags.length} tags provided.")
+        (tags.take(rs1.molecularBarcodeSegments.length), tags.drop(rs1.molecularBarcodeSegments.length))
+      }
+      else
+      {
+        (Seq.empty, Seq.empty)
+      }
+  }
```

Comment:
do you want to validate that the tags for the index bases does not overlap the tags for the index qualities?